zebra-network/src/address_book.rs

```diff
@@ -304,6 +304,8 @@ impl AddressBook {
         self.by_addr
             .values()
             .filter(|peer| peer.is_ready_for_connection_attempt())
```

Comment:
Bug Risk: Should we make this new check part of `is_ready_for_connection_attempt`, so it's used everywhere we check for connection attempts?

Or is `is_ready_for_connection_attempt` used for different purposes elsewhere?

Reply:
Done in [01d63d1](https://github.com/ZcashFoundation/zebra/pull/3030/commits/01d63d109500baa7056c8200983c78de4749dd3e). Yeah, I think it makes sense to move where the method is called. I also updated the documentation to describe the security implications a bit.

I checked for other usages, and I only found [`AddressBook::maybe_connected_peers`](https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/address_book.rs#L324-L331). I think that method might need to be improved in the future.

Reply:
The output of `AddressBook::maybe_connected_peers` won't change, because `Failed` peers are never connected.

Also, the maybe connected time is a few minutes, but the `is_probably_reachable` time is days, so the times never overlap either.